unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/PassportNavigationComponentController.cs

```diff
@@ -92,21 +107,59 @@ private void LoadAndShowOwnedWearables(UserProfile userProfile)
                                       var containedWearables = wearables.GroupBy(i => i.id).Select(g => g.First()).Take(MAX_NFT_COUNT)
                                          .Where(wearable => wearableCatalogBridge.IsValidWearable(wearable.id));
                                       view.SetCollectibleWearables(containedWearables.ToArray());
+                                      view.SetCollectibleWearablesLoadingActive(false);
                                   })
                                  .Catch(Debug.LogError);
         }
 
         private void LoadAndShowOwnedEmotes(UserProfile userProfile)
         {
+            view.SetCollectibleEmotesLoadingActive(true);
             emotesCatalogService.RequestOwnedEmotes(userProfile.userId)
                                  .Then(emotes =>
                                   {
                                       WearableItem[] emoteItems = emotes.GroupBy(i => i.id).Select(g => g.First()).Take(MAX_NFT_COUNT).ToArray();
                                       view.SetCollectibleEmotes(emoteItems);
+                                      view.SetCollectibleEmotesLoadingActive(false);
                                   })
                                  .Catch(Debug.LogError);
         }
 
+        private async UniTask LoadAndShowOwnedNamesAsync(UserProfile userProfile)
+        {
+            view.SetCollectibleNamesLoadingActive(true);
+            var ct = new CancellationTokenSource().Token;
+            var pagePointer = namesService.GetPaginationPointer(userProfile.userId, MAX_NFT_COUNT, ct);
+            var response = await pagePointer.GetPageAsync(1, CancellationToken.None);
+            var namesResult = new NamesResponse.NameEntry[] { };
+
+            if (response.success)
+                namesResult = response.response.Names.ToArray();
+            else
+                Debug.LogError("Error requesting names lambdas!");
+
+            view.SetCollectibleNames(namesResult);
+            view.SetCollectibleNamesLoadingActive(false);
+        }
+
+        private async UniTask LoadAndShowOwnedLandsAsync(UserProfile userProfile)
+        {
+            view.SetCollectibleLandsLoadingActive(true);
+            var ct = new CancellationTokenSource().Token;
+            // TODO (Santi): Use userProfile.userId here!!
+            var pagePointer = landsService.GetPaginationPointer(userProfile.userId, MAX_NFT_COUNT, ct);
+            var response = await pagePointer.GetPageAsync(1, CancellationToken.None);
+            var landsResult = new LandsResponse.LandEntry[] { };
+
+            if (response.success)
+                landsResult = response.response.Lands.ToArray();
+            else
+                Debug.LogError("Error requesting lands lambdas!");
+
+            view.SetCollectibleLands(landsResult);
+            view.SetCollectibleLandsLoadingActive(false);
+        }
+
```

Comment:
this also feels to me that some proper abstraction for emotes and wearables was missed here. Because logic is the same and only details+names are different. By using interface, some common interface and strategy pattern for the differences would be much better here

Reply:
I agree on some parts of the code, but i think that the gain is not enough for the effort that may take. Making abstractions would probably make it harder to follow, but having specific feature code, seems kind of redundant sometimes but its easier to follow and provides easier customization. However if it becomes a problem of maintenance over time we can try refactoring. What do you think?

Reply:
I would not agree in this particular, because it will be super easy to miss something when adding or correcting this code for `LoadAndShowOwnedLandsAsync` and `LoadAndShowOwnedNamesAsync`. I see same pattern there...
![image](https://user-images.githubusercontent.com/35366872/211667948-12cdbece-3701-4c30-8dc4-4c0a81377642.png)

Reply:
and also in SetCollecbleLand-Names, differences could be hidden via passing strategy-method ("functional" strategy pattern)
![image](https://user-images.githubusercontent.com/35366872/211668403-51f1d614-ff28-4fbc-8e1e-4ea56addfeb0.png)

Reply:
however. I will summon @AjimenezDCL and @mikhail-dcl again and leave it for you guys to decide where to do DRY and where AHA

Reply:
cc @Kinerius (also here)